### PR TITLE
Emr master no public ip changes done to boilerplate code as per pylint

### PR DESCRIPTION
**Pull request by almasshk**
I confirm these files are made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)

*Issue #, if available:*

*Description of changes:*

### CODE REVIEW

**Comment by jongogogo** [comment]
Please don't remove the `pylintrc`

File: python/EMR_MASTER_NO_PUBLIC_IP/parameters.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "Version": "1.0",
+  "Parameters": {
+    "RuleName": "EMR_MASTER_NO_PUBLIC_IP",
+    "SourceRuntime": "python3.6",
+    "CodeKey": "EMR_MASTER_NO_PUBLIC_IP.zip",
+    "InputParameters": "{}",
+    "OptionalParameters": "{}",
+    "SourceEvents": "AWS::EC2::Instance",
```
Comment (jongogogo) [inline python/EMR_MASTER_NO_PUBLIC_IP/parameters.json]: Remove this line. This is for configuration changes.

File: python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP.py [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+    if cluster_dns_list:
+        for cluster in cluster_dns_list:
+            #Scenario 3: DescribeInstances doesn't have public DNS for the master node of the cluster while ListInstances has it.
+            if not cluster[1]:
+                evaluations.append(build_evaluation(cluster[0],
+                                                    'COMPLIANT',
+                                                    event))
+            #Scenario 2: Both DescribeInstances and ListInstances have public DNS for the master node of the cluster.
+            else:
+                evaluations.append(build_evaluation(cluster[0],
```
Comment (jongogogo) [inline python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP.py]: Can you put everything on the same line? I think it is more readable.

File: python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP.py [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+    private_dns_cluster_list, cluster_dns_list = cluster_dns_mapping(cluster_list, emr_client, ec2_client)
+
+    #cluster_dns_list is a list of lists which has cluster id and public DNS mappings
+    #If public DNS is empty then Compliant else not Compliant
+
+    if cluster_dns_list:
+        for cluster in cluster_dns_list:
+            #Scenario 3: DescribeInstances doesn't have public DNS for the master node of the cluster while ListInstances has it.
+            if not cluster[1]:
+                evaluations.append(build_evaluation(cluster[0],
```
Comment (jongogogo) [inline python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP.py]: Can you put everything on the same line? I think it is more readable.

File: python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP.py [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+
+    emr_client = get_client('emr', event)
+
+    #Listing RUNNING and WAITING clusters
+    cluster_list = list_all_clusters(emr_client)
+    #Scenario 1: If no RUNNING and WAITING clusters then return NOT_APPLICABLE
+    if not cluster_list:
+        return None
+
+    #Building EC2 service client using get_client()
```
Comment (jongogogo) [inline python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP.py]: Remove, comments need to be meaningful. If it is obvious please remove.

File: python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP_test.py [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+        resp_expected.append(build_expected_response('COMPLIANT',
+                                                     compliance_resource_id='j-AAAAA0AAAAA'))
+
+        response = RULE.lambda_handler(build_lambda_scheduled_event(), {})
+        assert_successful_evaluation(self, response, resp_expected)
+
+    #Scenario 4: The ListInstances call doesn't have public DNS for the master node of the cluster.
+    #Test for when the cluster is in RUNNING state and master node as no public IP, COMPLIANT
+    def test_scenario_4_running_private_cluster(self):
+        listcluster_valid_running = {'Clusters': [{'Id': 'j-AAAAA0AAAAA', 'Status': {'State': 'RUNNING'}}]}
```
Comment (jongogogo) [inline python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP_test.py]: Please test on at least 2 clusters.

File: python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP_test.py [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+        resp_expected = []
+        resp_expected.append(build_expected_response('NOT_APPLICABLE',
+                                                     compliance_resource_id='123456789012',
+                                                     compliance_resource_type="AWS::::Account"))
+        assert_successful_evaluation(self, response, resp_expected)
+
+    #Scenario 2: Both DescribeInstances and ListInstances have public DNS for the master node of the cluster.
+    #Test for when the cluster is in RUNNING state and master node as a public IP, NON-COMPLIANT
+    def test_scenario_2_running_public_cluster(self):
+        listcluster_valid_running = {'Clusters': [{'Id': 'j-AAAAA0AAAAA', 'Status': {'State': 'RUNNING'}}]}
```
Comment (jongogogo) [inline python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP_test.py]: Please test on at least 2 clusters.

File: python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP_test.py [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+                                                     compliance_resource_id='j-AAAAA0AAAAA',
+                                                     annotation="The master node of the EMR cluster has a public IP."))
+        response = RULE.lambda_handler(build_lambda_scheduled_event(), {})
+        assert_successful_evaluation(self, response, resp_expected)
+
+
+    #Scenario 3: DescribeInstances doesn't have public DNS for the master node of the cluster while ListInstances has it.
+    #Test for when the cluster is in a Private Subnet but EIP was attached and then detached from the master node
+    def test_scenario_3_running_private_cluster_eip_removed(self):
+        listcluster_valid_running = {'Clusters': [{'Id': 'j-AAAAA0AAAAA', 'Status': {'State': 'RUNNING'}}]}
```
Comment (jongogogo) [inline python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP_test.py]: Please test on at least 2 clusters.

**Comment by almasshk** [comment]
incorporated comments

**Comment by jongogogo** [comment]
Appear to be duplicate of https://github.com/awslabs/aws-config-rules/pull/193

**Pull request closed without merging**